### PR TITLE
Pig

### PR DESCRIPTION
**Pull request by akshaisarma**

### CODE REVIEW

**Comment by akshaisarma** [comment]
Code coverage is broken because of Cobertura's issues with Java 8. I will look into alternatives or get a clover license for Open Source.

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by akshaisarma** [comment]
Added clover and applied for a license. Coveralls doesn't support clover yet unfortunately but has an issue for this at trautonen/coveralls-maven-plugin#36

**Comment by akshaisarma** [comment]
Adding some unrelated unit tests. Clover reports coverage at 96.3. It was around 60 before. The coveralls number of 99 was not accurate.

**Comment by akshaisarma** [comment]
I added Jacoco instead of cobertura. It works because actual coverage report of 96% caused the check to fail. I also bound checkstyle and jacoco to the verify and test lifecycles and temporarily added a line coverage requirement of 95% and branch of 90%. There is one more class that is not tested well - App.

I also have an open source clover license from Atlassian. I find that using clover for local coverage visualization is better. Please reach out to me for the license if you need it.

**Comment by akshaisarma** [comment]
Coverage is now 99.5. I could have hit 100 but would require sacrificing brevity in TypeSystem.Operations.

File: src/main/java/com/yahoo/validatar/execution/hive/Apiary.java [hunk 1]
```diff
@@ -38,11 +38,13 @@
 import static java.util.Collections.singletonList;
 
 public class Apiary implements Engine {
+    public static final String HIVE_JDBC = "hive-jdbc";
```
Comment (joshwalters) [inline src/main/java/com/yahoo/validatar/execution/hive/Apiary.java]: Nice, thanks for pulling all these string constants out. Much cleaner.

File: src/main/java/com/yahoo/validatar/execution/pig/Sty.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        public FieldDetail(String alias, byte type) {
+            this.alias = alias;
+            this.type = type;
+        }
+    }
+
+    private OptionParser parser = new OptionParser() {
+        {
+            acceptsAll(singletonList(PIG_EXEC_TYPE), "The exec-type for Pig to use.  This is the " +
+                                                       "-x argument used when running Pig. Ex: local, mr, tez etc. ")
```
Comment (joshwalters) [inline src/main/java/com/yahoo/validatar/execution/pig/Sty.java]: Cool, so we could execute pig script Validatar tests against sample data locally without any hadoop cluster!

File: src/main/java/com/yahoo/validatar/execution/pig/Sty.java [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+                return TypeSystem.asTypedObject(new Timestamp(DataType.toDateTime(data, type).getMillis()));
+            case DataType.BYTE:
+            case DataType.BYTEARRAY:
+            case DataType.CHARARRAY:
+                return TypeSystem.asTypedObject(DataType.toString(data, type));
+            case DataType.BIGINTEGER:
+            case DataType.BIGDECIMAL:
+                return TypeSystem.asTypedObject(DataType.toBigDecimal(data, type));
+            default:
+                //TUPLE, BAG, MAP, INTERNALMAP, GENERIC_WRITABLECOMPARABLE, ERROR, UNKNOWN, NULL and anything else
```
Comment (joshwalters) [inline src/main/java/com/yahoo/validatar/execution/pig/Sty.java]: Pig queries that return a map/tuple/etc wont be parseable by Validadar (at least for now)? They are unlikely to be used for these sort of validation queries anyways.

**Comment by joshwalters** [comment]
This looks good to me, awesome stuff!

File: src/main/java/com/yahoo/validatar/execution/pig/Sty.java (hunk 3 above)
Comment (akshaisarma) [inline src/main/java/com/yahoo/validatar/execution/pig/Sty.java]: That's right - unless we want to support extracting fields from them or allow operators to work on complex fields. I think we should push all of that extraction logic to the engine itself and require the user to pull out atomic data.

**Comment by akshaisarma** [comment]
I do want to test against a Hadoop cluster and perhaps write some documentation on what Pig/Hadoop jars and resources are needed to run it. I also want to test if registering UDFs etc in the script will work. I will hold off merging till that is cleared up. We may also need more settings that PigMain sets by default, notably tmpfilecompression etc.

**Pull request merged**